pkg/cluster/install.go

```diff
@@ -48,7 +48,13 @@ func (m *manager) AdminUpdate(ctx context.Context) error {
 
 func (m *manager) Update(ctx context.Context) error {
 	steps := []steps.Step{
+		steps.Action(m.initializeKubernetesClients), // must be first
 		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.validateResources)),
+		// credentials rotation flow steps
+		steps.Action(m.createOrUpdateClusterServicePrincipalRBAC),
+		steps.Action(m.createOrUpdateDenyAssignment),
```

Comment:
Currently these 2 are almost impossible to test in units. This is because we call `m.clusterSPObjectID(ctx)` which on its own calls aad, autorest, graphrback. Tempted to create a separate post merge story to move this methods outside, mock it all together and do tests?

Reply:
https://github.com/Azure/ARO-RP/pull/1316 .  I think this should probably be rebased on top.